LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,109 @@ struct OpenAIChatView: View {
+            ChatFunctionDeclaration(
+                name: "get_resource_titles",
+                description: String(localized: "MULTIPLE_RESOURCE_FUNCTION_DESCRIPTION"),
+                parameters: JSONSchema(
+                    type: .object,
+                    properties: [
+                        "resources": .init(
+                            type: .string,
+                            description: String(localized: "MULTIPLE_RESOURCE_PARAMETER_DESCRIPTION"),
+                            enumValues: stringResourcesArray
+                        )
+                    ],
+                    required: ["resources"]
+                )
+            )
+        ]
+        
+        let query = ChatQuery(
+            model: "gpt-3.5-turbo-0613",
+            messages: [
+                Chat(role: .user, content: chat.last?.content ?? ""),
+                Chat(role: .system, content: String(localized: "MULTIPLE_RESOURCE_FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+            ],
+            functions: functions
+        )
+        
+        let result = try await openAI.chats(query: query)
+        
+        var functionCallOutputArray = [String]()
+        
+        let rawFunctionCallOutput = result.choices.rawValue
+        if let rawData = rawFunctionCallOutput.data(using: .utf8) {
+            if let jsonArray = try? JSONSerialization.jsonObject(with: rawData, options: .allowFragments) as? [[String: Any]] {
+                for jsonDictionary in jsonArray {
+                    guard let finishReason = jsonDictionary["finish_reason"] as? String,
+                          let messageDict = jsonDictionary["message"] as? [String: Any],
+                          let functionCallDict = messageDict["function_call"] as? [String: Any],
+                          let arguments = functionCallDict["arguments"] as? String,
+                          let data = arguments.data(using: .utf8),
+                          let argumentsJson = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
+                          let resources = argumentsJson["resources"] as? String else {
+                        print("Error extracting data")
+                        continue
+                    }
+                    
+                    functionCallOutputArray = resources.components(separatedBy: ",")
+                    print("Resources: \(functionCallOutputArray)")
+                }
+            } else {
+                print("Error extracting data")
+            }
+        }
+
+        return functionCallOutputArray
+    }
+
+    private func processFunctionCallOutputArray(functionCallOutputArray: [String], resourcesArray: [FHIRResource]) {
+        var stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        for resource in functionCallOutputArray {
+            var stringResource = resource
+            stringResource = resource.trimmingCharacters(in: .whitespaces)
```

Comment:
Can be combined in a single statement.
```suggestion
            let stringResource = resource.trimmingCharacters(in: .whitespaces)
```